lib/mvi/service.rb

```diff
@@ -33,6 +33,34 @@ class Service < Common::Client::Base
     STATSD_KEY_PREFIX = 'api.mvi' unless const_defined?(:STATSD_KEY_PREFIX)
     SERVER_ERROR = 'server_error'
 
+    # rubocop:disable Metrics/MethodLength
+    def add_person(user_identity)
+      with_monitoring do
+        measure_info(user_identity) do
+          raw_response = perform(
+            :post, '',
+            create_add_message(user_identity),
+            soapaction: OPERATIONS[:add_person]
+          )
+          MVI::Responses::AddPersonResponse.with_parsed_response(raw_response)
+        end
+      end
+    rescue Breakers::OutageException => e
+      Raven.extra_context(breakers_error_message: e.message)
+      log_console_and_sentry('MVI add_person connection failed.', :warn)
+      mvi_add_exception_response_for('MVI_503', e)
+    rescue Faraday::ConnectionFailed => e
+      log_console_and_sentry("MVI add_person connection failed: #{e.message}", :warn)
+      mvi_add_exception_response_for('MVI_504', e)
+    rescue Common::Client::Errors::ClientError, Common::Exceptions::GatewayTimeout => e
+      log_console_and_sentry("MVI add_person error: #{e.message}", :warn)
+      mvi_add_exception_response_for('MVI_504', e)
+    rescue MVI::Errors::Base => e
+      mvi_error_handler(user_identity, e)
+      mvi_add_exception_response_for('MVI_502', e)
+    end
+    # rubocop:enable Metrics/MethodLength
```

Comment:
Looks like a good place to extract this error handling chain to a private method (or middleware if you're so inclined)